### PR TITLE
[coop] Generate native-to-managed thread attaching in marshal + Reset mono_jit_thread_attach signature

### PR DESCRIPTION
**Pull request by ludovic-henry**

### CODE REVIEW

**Comment by ludovic-henry** [comment]
@rolf that reverts the behaviour and signature of `mono_jit_thread_attach` to fix XM. `mono_threads_attach_coop` / `mono_threads_detach_coop` is what you should use for watchos now.

@kumpera @vargaz could you please review?

File: mono/mini/method-to-ir.c [hunk 1]
```diff
@@ -12973,6 +12906,12 @@ mono_method_to_ir (MonoCompile *cfg, MonoMethod *method, MonoBasicBlock *start_b
 				constrained_class = NULL;
 				break;
 			}
+			case CEE_MONO_LDDOMAIN:
+				CHECK_STACK_OVF (1);
+				EMIT_NEW_PCONST (cfg, ins, cfg->compile_aot ? NULL : cfg->domain);
```
Comment (alexrp) [inline mono/mini/method-to-ir.c]: Should this not be using `EMIT_NEW_DOMAINCONST (cfg, ins)`?

File: mono/metadata/sgen-mono.c [hunk 2]
```diff
@@ -2253,7 +2253,7 @@ static void
 sgen_thread_detach (SgenThreadInfo *p)
 {
 	/* If a delegate is passed to native code and invoked on a thread we dont
-	 * know about, the jit will register it with mono_jit_thread_attach, but
+	 * know about, marshal will register it with mono_marshal_thread_attach, but
```
Comment (lambdageek) [inline mono/metadata/sgen-mono.c]: What's `mono_marshal_thread_attach`?

Comment (ludovic-henry) [inline mono/metadata/sgen-mono.c]: How I named `mono_threads_attach_coop` before, let me fix that.

**Pull request merged**